yt_dlp/extractor/redgifs.py

```diff
@@ -158,56 +158,59 @@ def _real_extract(self, url):
 
 class RedGifsSearchIE(RedGifsBaseInfoExtractor):
     IE_DESC = 'Redgifs search'
-    _VALID_URL = r'https?://(?:www\.)?redgifs\.com/browse\?(?P<query>[^#]+)'
+    _VALID_URL = r'https?://(?:www\.)?redgifs\.com/gifs/(?P<tags>[^?/]+)(\?(?P<query>[^#]+))?'
     _PAGE_SIZE = 80
     _TESTS = [
         {
-            'url': 'https://www.redgifs.com/browse?tags=Lesbian',
+            'url': 'https://www.redgifs.com/gifs/lesbian?tab=gifs',
             'info_dict': {
-                'id': 'tags=Lesbian',
-                'title': 'Lesbian',
-                'description': 'RedGifs search for Lesbian, ordered by trending',
+                'id': 'search_text=lesbian&order=trending&type=g',
+                'title': 'lesbian',
+                'description': 'RedGifs search for lesbian, ordered by trending',
             },
             'playlist_mincount': 100,
         },
         {
-            'url': 'https://www.redgifs.com/browse?type=g&order=latest&tags=Lesbian',
+            'url': 'https://www.redgifs.com/gifs/lesbian?order=new&tab=gifs',
             'info_dict': {
-                'id': 'type=g&order=latest&tags=Lesbian',
-                'title': 'Lesbian',
-                'description': 'RedGifs search for Lesbian, ordered by latest',
+                'id': 'search_text=lesbian&order=new&type=g',
+                'title': 'lesbian',
+                'description': 'RedGifs search for lesbian, ordered by latest',
             },
             'playlist_mincount': 100,
         },
-        {
-            'url': 'https://www.redgifs.com/browse?type=g&order=latest&tags=Lesbian&page=2',
-            'info_dict': {
-                'id': 'type=g&order=latest&tags=Lesbian&page=2',
-                'title': 'Lesbian',
-                'description': 'RedGifs search for Lesbian, ordered by latest',
-            },
-            'playlist_count': 80,
-        },
```

Comment:
The current search page is an "endless scroll" thing, without it having a page parameter (only the API does).

Example (obviously nsfw): https://www.redgifs.com/gifs/lesbian?order=top28&tab=gifs